pkg/experiment/metastore/metastore.go

```diff
@@ -233,7 +247,13 @@ func (m *Metastore) initRaft() (err error) {
 		_ = level.Info(m.logger).Log("msg", "restoring existing state, not bootstraping")
 	}
 
-	m.leaderhealth.Register(m.raft, metastoreRaftLeaderHealthServiceName)
+	m.leaderhealth.Register(m.raft, func(st raft.RaftState) {
+		if st == raft.Leader {
+			m.dlq.Start()
+		} else {
+			m.dlq.Stop()
+		}
+	})
```

Comment:
I'm not sure we want to use `leaderhealth` now that we do not use grpc health thing for leader discovery. I'd suggest removing it altogether.

On a side note: this assumes that DLQ recovery may run on a follower (as this is a weak check), which should be fine as long as we ensure idempotence of the `AddBlock` call. Is this correct?

Reply:
> I'd suggest removing it altogether.

The health related bits were removed and I removed the word health from type names. Let me know if this is sufficient. We still need to observe leadership status to start/stop recovery routine